common/types/data_connections.ts

```diff
@@ -124,6 +124,7 @@ export interface CachedDataSource {
 }
 
 export interface DataSourceCacheData {
+  dataSourceMDSId: string;
```

Comment:
I think this is not backward compatible: 
* Either this should be made optional 
* Or we should update the cache structure from version for v1 to v2 and have code to support both.